src/Objs/Homepage/HomepageEditingConfig.ts

```diff
@@ -128,5 +128,32 @@ provideEditingConfig(Homepage, {
     siteFontHeadlineWeight: '500',
     siteRoundedCorners: true,
   },
-  validations: defaultPageValidations,
+  validations: [
+    ...defaultPageValidations,
+    [
+      '_language',
+      (language: string | null, { obj }) => {
+        if (!language) {
+          return {
+            message: 'The language must be set.',
+            severity: 'error',
+          }
+        }
+
+        const contentId = obj.contentId()
+        const duplicates = Obj.onAllSites()
+          .where('_objClass', 'equals', 'Homepage')
+          .and('_contentId', 'equals', contentId)
+          .and('_language', 'equals', language)
+          .count()
```

Comment:
* [ ] FYI How about iterating over `versionsOnAllSites()`? This may look less efficient but the result is already in memory b/c of the language switch logic.